src/adapters/scene.js

```diff
@@ -107,28 +106,49 @@ Scene.prototype.initMapBox = function() {
         this.mb.getCanvas().style.cursor = '';
       });
 
+      // The OSM PoIs are on separate interactive layers.
```

Comment:
After fiddling with tests, I think we could go one step further with this simplification, to get rid of this `_mapPoi` hack.

It makes both listeners (the interactive one and the generic one) "communicate", but it is far too fragile. It relies on the assumption that it's the same event object used by the MapBox-GL implementation to trigger both listenernes, but nothing guarantees that and it could break in the future.

By leveraging `queryRenderedFeatures`, it's possible to do everything in a single, generic listener, making everything simpler and more solid.

```js
this.mb.on('click', e => {
      const features = this.mb.queryRenderedFeatures(e.point, { layers: interactiveLayers });

      const poi = features.length > 0 ? new MapPoi(features[0]) : new LatLonPoi(e.lngLat);

      // do stuff with the POI
```

As this is called only once in a while, I don't think there are performance issues to use `queryRenderedFeatures` instead of direct listeners on the layer themselves.

Sadly this doesn't fix the tests yet, but at least we would have only one event to fake, not two.
I think this could be done in a separate PR, to be merged before this one. We would address this test problem in it. A really simple solution for that is to call a method `clickOnPoi()` instead of doing everything directly in the event listener, which requires faking events, which are badly supported by the mock.

What do you think? @xem @amatissart